Add possibility to install a custom compiled slurm package

ahmam opened the pull request:
Add the possibility to install custom slurm packages, the objective to install the most recent vesrion slurm on debian or ubuntu.

ahmam committed:
Add new custom services (#2)

ahmam committed:
install custom slurm packages (#1)

* Update main.yml

* Update defaults/main.yml

Co-authored-by: Bruno Travouillon <devel@travouillon.fr>

* install custom packages

* install custom packages

* install custom packages

* add custom debain repositories

* install custom packages

* install custom packages

* fix typo dynamique vars

* fix the FQCN

* add configure custom repos

* add example of custom repo

* add new  README.md

Co-authored-by: Bruno Travouillon <devel@travouillon.fr>

* remove custom package installtion

* remove condition slurm_apt_repository

* fix typo

* remove slurm_apt_repository condition

* remove slurm_apt_repository condition

* fix custom log directory creation

* move var to first part

* rename the var to static

* fix var

---------

Co-authored-by: Bruno Travouillon <devel@travouillon.fr>
Co-authored-by: ahmed <you@example.com>

phuelsdunk committed:
Added template for plugstack.conf

cat-bro committed:
Remove key 'SlurmctldPidFile' from __slurmdbd_config_default

(cherry picked from commit 936e64adf0a9d4bbc5641bf8e4471db73c9037cb)

clil16 committed:
Fix typo in handler name

(cherry picked from commit 712cf3278112ff97d6e40694d5172fd4701efc6d)

btravouillon committed:
Merge remote-tracking branch 'upstream/main' into mila

btravouillon committed:
Allow to define several SlurmctldHost in a list (#4)

SlurmctldHost can be defined more than once in slurm.conf to define a
primary and backups Slurm controllers. If SlurmctldHost is defined as a
list in `slurm_config`, then the template will add one line per list
item in slurm.conf..

btravouillon committed:
Make sure slurm_start_services is a boolean

When ansible-playbook is called with `-e slurm_start_services=false`,
the variable is evaluated as a string (which is true). Apply the
`| bool` filter to evaluate the variable as a boolean.

btravouillon committed:
Do not setup a new SlurmDBD cluster by default

While the new feature to create a new SlurmDBD cluster could be nice in
some cases, it causes some issues when one wants to reinstall a node
with an existing database. IMHO, it is safer to disable this by default.
One should opt-in for Ansible to create a new cluster.

btravouillon committed:
Add support for job_container.conf

btravouillon committed:
Restart slurm services after a Slurm upgrade

Make sure to execute the correct Slurm after the upgrade. Otherwise,
this may cause issues when reloading the service during logrotate since
some files changed on the system.

btravouillon committed:
Restart slurmd and slurmctld after a Slurm upgrade

Make sure to execute the correct Slurm after the upgrade. Otherwise,
this may cause issues when reloading the service during logrotate since
some files changed on the system.

btravouillon commented on tasks/main.yml:
```diff
@@ -3,6 +3,10 @@
 - name: Include user creation tasks
   ansible.builtin.include_tasks: user.yml
   when: slurm_create_user
+ 
+- name: Include Configure custom repositories
+  ansible.builtin.include_tasks: repositories-Debian.yml
+  when: slurm_configure_repos
```
Should this support RHEL-like distros as well?
At least, this task should not execute when a host's family OS is not Debian.

btravouillon committed:
Skip service reload when restart already triggered

Upon changes, some tasks will trigger the reload handler while others
will trigger the restart handler. During initial installation of Slurm
with the role, this can lead to some race condition where reload and
restart will execute too closely one from the other, which can trigger a
failure of the second service.

When a restart handler is notified, there should be no need to trigger
the reload as well. Move the restart handlers before the reload handlers
to make sure those are executed first. Register their result in a
variable. Test if the variable is defined in the reload handler (which
means that the restart handler was executed).

Also remove one of the 'Reload slurmdbd' handler which was defined
twice.

btravouillon committed:
Update galaxy_info for import into mila namespace

btravouillon committed:
Merge remote-tracking branch 'upstream/main' into mila

btravouillon commented:
@ahmam I guess you should use another branch than `mila` for this PR since this is our soft-fork branch where we apply all our changes.

btravouillon committed:
Add support for job_submit.lua

btravouillon committed:
This commit adds support for Slurm's RebootProgram feature, allowing
automatic node rebooting through Slurm. The implementation includes:

- New reboot_program.j2 template that handles node rebooting using
  either systemd or traditional init systems
- Configuration variable slurm_reboot_program to enable/disable the
  feature
- Task to install and configure the reboot program script
- Documentation in README.md explaining how to use the feature

The reboot program includes logging of reboot attempts and proper error
handling. It will attempt to use systemd's reboot command first, falling
back to traditional shutdown command if systemd is not available.

btravouillon committed:
Merge remote-tracking branch 'upstream/main' into mila

btravouillon committed:
Fix logger text in reboot script

There is no arg passed to the reboot script.

Also update the README to make it more readable.

btravouillon committed:
Add support for Slurm Topology

Create a new variable slurm_topology_config to define the content of the
topology.conf file and ensure TopologyPlugin is defined in slurm.conf
when the topology must be configured.

btravouillon committed:
Install topology.conf on controllers and computes

The file topology.conf must exist when `TopologyPlugin` is defined in
slurm.conf, otherwise slurmd won't be able to start.

jsl6ul committed:
fix: apt_key module is deprecated, replace it with get_url & gpg (#19)
